OpenChange/MAPIStoreMailMessage.m

```diff
@@ -1394,6 +1394,28 @@ - (int) getPidLidMeetingType: (void **) data
           : MAPISTORE_ERR_NOT_FOUND);
 }
 
+- (int) getPidTagTransportMessageHeaders: (void **) data
+                    inMemCtx: (TALLOC_CTX *) memCtx
+{
+  NSDictionary *allHeaders = [sogoObject mailHeaders];
+  NSString *headersString = @"";
+  NSString *format= @"%@: %@\n";
+  int lastHeader = [allHeaders count] - 1;
```

Comment:
It would be much simple to add each item to an array and then use `[items componentsJoinedByString: "\n"]`